### PR TITLE
[Diffusion] Fuse ERNIE-Image Q/K RoPE for eager execution

### PR DESCRIPTION
**Pull request by dongbo910220**
### What does this PR do?

This PR reduces ERNIE-Image eager execution latency by fusing the Q and K rotate-half RoPE paths into one Triton launch. The fused path consumes the existing unexpanded cosine and sine tensors, avoids the repeated eager materialization chain, and preserves the original BF16 rounding behavior bit for bit.

The optimization is limited to supported CUDA inference inputs outside `torch.compile`. Compiled execution retains the native PyTorch expressions for Inductor, and unsupported inputs or failed verification use the existing implementation.

### Test

```bash
python -m pytest -q tests/diffusion/models/ernie_image/test_ernie_image_fused_rope.py tests/diffusion/models/ernie_image/test_ernie_image_rope.py -p no:cacheprovider
```

all passed

Measured on one NVIDIA GeForce RTX 5090 with `baidu/ERNIE-Image`, 1024x1024 output, five denoising steps, seed 42, one warmup, and five measured requests per arm:

| Metric | main | PR | Improvement |
| --- | ---: | ---: | ---: |
| Eager E2E median | 4.4431 s | 4.3047 s | 3.11% lower |
| Eager Q/K RoPE median, shape `[2, 4608, 32, 128]` | 1.1969 ms | 0.1941 ms | 6.17x |

### Design & Code Changes

- Add a model-local Triton kernel that computes Q and K RoPE in one launch and reproduces the eager BF16 rounding boundaries.
- Verify each new runtime geometry against the native implementation with `torch.equal`; mismatches or launch failures permanently fall back for that process.
- Keep gradients, unsupported devices, dtypes, layouts, and `torch.compile` on the native path without adding a new user-facing flag.

### CODE REVIEW

**Comment by dongbo910220** [comment]
Self-review completed. I checked the Triton indexing and BF16 rounding, the Q/K integration, supported-input gates, one-time bit-exact verification, fail-closed fallback, and preservation of the native `torch.compile` and autograd paths. I also reviewed the added parity and routing tests; targeted tests and the current CI checks all passed, and the benchmarked eager output remained byte-identical.

File: vllm_omni/diffusion/models/ernie_image/fused_rope.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    except Exception:
+        _FAILED_KEYS.add(runtime_key)
+        logger.exception("ERNIE-Image fused Q/K RoPE failed; using the eager fallback")
+        return None
+
+    if runtime_key in _VERIFIED_KEYS:
+        return output_query, output_key
+
+    eager_query = eager_fn(query, freqs_cos, freqs_sin)
+    eager_key = eager_fn(key, freqs_cos, freqs_sin)
```
Comment (lishunyang12) [inline vllm_omni/diffusion/models/ernie_image/fused_rope.py]: Move bit-exact verification into tests; don't synchronize inference for every new shape.

File: vllm_omni/diffusion/models/ernie_image/fused_rope.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+from collections.abc import Callable
+
+import torch
+from vllm.triton_utils import HAS_TRITON, tl, triton
+
+from vllm_omni.platforms import current_omni_platform
+
+logger = logging.getLogger(__name__)
+
+_FAILED_KEYS: set[tuple[object, ...]] = set()
```
Comment (lishunyang12) [inline vllm_omni/diffusion/models/ernie_image/fused_rope.py]: Bound both runtime-key caches; variable request shapes currently accumulate forever.

File: vllm_omni/diffusion/models/ernie_image/fused_rope.py (hunk 1 above)
Comment (dongbo910220) [reply vllm_omni/diffusion/models/ernie_image/fused_rope.py]: Moved bit-exact validation entirely into the CUDA tests. The inference path now returns the fused result directly for supported inputs, while the tests call the fused helper directly, assert that it does not fall back, and compare both Q and K against the native implementation with `torch.equal` for representative and production shapes.

File: vllm_omni/diffusion/models/ernie_image/fused_rope.py (hunk 2 above)
Comment (dongbo910220) [reply vllm_omni/diffusion/models/ernie_image/fused_rope.py]: Removed the verified-shape cache together with runtime verification, and changed the remaining failed-shape cache to a 128-entry LRU. I also added coverage for capacity and oldest-entry eviction.

**Comment by lishunyang12** [comment]
show visual comparision

**Comment by dongbo910220** [comment]
@lishunyang12 Thanks — here is the visual comparison using the same frozen prompt and seed for the baseline and the current PR. The generated 1024×1024 RGB outputs are byte-identical (`0 / 3,145,728` differing channel values; maximum absolute difference `0`).

Baseline and current PR visual comparison

**Review by lishunyang12 — APPROVED** [review]
